File: Aio-Algorithms-And-Solutions/Solutions/Exercism/CSharp/ResistorColorDuo.cs
namespace Aio_Algorithms_And_Solutions.Solutions.Exercism.CSharp;

using System;
using System.Collections.Generic;

public static class ResistorColorDuo
{
    private static readonly Dictionary<string, int> ColorValues = new Dictionary<string, int>
    {
                    ["black"] = 0,
                    ["brown"] = 1,
                    ["red"] = 2,
                    ["orange"] = 3,
                    ["yellow"] = 4,
                    ["green"] = 5,
                    ["blue"] = 6,
                    ["violet"] = 7,
                    ["grey"] = 8,
                    ["white"] = 9
    };

    public static int Value(string[] colors)
    {
        if (colors.Length < 2)
        {
            throw new ArgumentException("At least two colors required.");
        }

        return 10 * ColorValues[colors[0]] + ColorValues[colors[1]];
    }
}

File: Aio-Algorithms-And-Solutions/Solutions/Exercism/Java/ResistorColorDuo.java
package exercism;

class ResistorColorDuo {
    private static final String[] COLORS = {
            "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
    };

    int value(String[] colors) {
        int value = 0;
        for (int i = 0; i < 2 && i < colors.length; i++) {
            for (int j = 0; j < COLORS.length; j++) {
                if (COLORS[j].equals(colors[i])) {
                    value = value * 10 + j;
                    break;
                }
            }
        }
        return value;
    }
}

